nautobot/extras/tests/test_customfields.py

```diff
@@ -385,15 +385,51 @@ def test_to_filter_field(self):
 
 class CustomFieldManagerTest(TestCase):
     def setUp(self):
-        content_type = ContentType.objects.get_for_model(Location)
+        self.content_type = ContentType.objects.get_for_model(Location)
         custom_field = CustomField(type=CustomFieldTypeChoices.TYPE_TEXT, label="Text Field", default="foo")
         custom_field.save()
-        custom_field.content_types.set([content_type])
+        custom_field.content_types.set([self.content_type])
 
     def test_get_for_model(self):
         self.assertEqual(CustomField.objects.get_for_model(Location).count(), 2)
         self.assertEqual(CustomField.objects.get_for_model(VirtualMachine).count(), 0)
 
+    def test_get_for_model_lru_cache(self):
+        """Test that the lru cache is properly invalidated when CustomFields are created or deleted."""
+
+        qs1 = CustomField.objects.get_for_model(Location)
+        self.assertIsNone(qs1._result_cache)
+        list(qs1)  # force queryset evaluation
+        self.assertIsNotNone(qs1._result_cache)
+
+        # Assert that the cache is invalidated on object save
```

Comment:
Should we have at least one test asserting that the cache is used when no change has been made?

Reply:
I actually had this before but removed it because it made the test difficult to read. I'll write another test for that.